core/src/test/java/org/apache/iceberg/rest/TestRESTCatalog.java

```diff
@@ -2655,6 +2656,68 @@ public void testCleanupCleanableExceptionsReplace() {
         .isInstanceOf(NotFoundException.class);
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {"1", "2"})
+  public void testCreateTableTransaction(String formatVersion) {
+    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
+    RESTCatalog catalog = catalog(adapter);
+
+    if (requiresNamespaceCreate()) {
+      catalog.createNamespace(NS);
+    }
+
+    // Test create table transaction with no data file
+    Transaction createTableTransaction1 =
+        catalog.newCreateTableTransaction(
+            TableIdentifier.of(NS, "table1"),
+            SCHEMA,
+            PartitionSpec.unpartitioned(),
+            ImmutableMap.of("format-version", formatVersion));
+    assertThatNoException().isThrownBy(() -> createTableTransaction1.commitTransaction());
```

Comment:
rather than checking that no exception was thrown it would be better to just load the table and ensure it exists.

Something like this short piece should be enough:

```
@ParameterizedTest
  @ValueSource(ints = {1, 2})
  public void createTableTransaction(int formatVersion) {
    RESTCatalogAdapter adapter = Mockito.spy(new RESTCatalogAdapter(backendCatalog));
    RESTCatalog catalog = catalog(adapter);

    if (requiresNamespaceCreate()) {
      catalog.createNamespace(NS);
    }

    catalog
        .newCreateTableTransaction(
            TABLE,
            SCHEMA,
            PartitionSpec.unpartitioned(),
            ImmutableMap.of("format-version", String.valueOf(formatVersion)))
        .commitTransaction();

    BaseTable table = (BaseTable) catalog.loadTable(TABLE);
    assertThat(table.operations().current().formatVersion()).isEqualTo(formatVersion);
  }
```
Can you please also update the other test accordingly?

Reply:
In fact we should move this to `CatalogTests`, so that we make sure this works across different catalog implementations:

```
@ParameterizedTest
  @ValueSource(ints = {1, 2})
  public void createTableTransaction(int formatVersion) {
    if (requiresNamespaceCreate()) {
      catalog().createNamespace(NS);
    }

    catalog()
            .newCreateTableTransaction(
                    TABLE,
                    SCHEMA,
                    PartitionSpec.unpartitioned(),
                    ImmutableMap.of("format-version", String.valueOf(formatVersion)))
            .commitTransaction();

    BaseTable table = (BaseTable) catalog().loadTable(TABLE);
    assertThat(table.operations().current().formatVersion()).isEqualTo(formatVersion);
  }
```

Reply:
Tests have been moved to `CatalogTests`, and modified as your suggestion. Please take a look when available, thanks a lot.